2025-05-09 20:02:19,859 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":324,"user":3,"application":9,"start_time":"2025-05-09T20:02:18.806438+03:00","end_time":"2025-05-09T20:02:19.807828+03:00","duration":"00:00:01.001390","keyboard_presses":0}
2025-05-09 20:02:19,860 - TimeTracker - INFO - Успешно отправлено 4 записей активности.
2025-05-09 20:02:20,364 - TimeTracker - INFO - Обновлен список приложений: 401 процессов
2025-05-09 20:02:20,371 - TimeTracker - INFO - Получено 7 приложений с сервера
2025-05-09 20:02:20,811 - TimeTracker - INFO - Завершена сессия активности: App='python.exe', Title='Time Tracker PRO, Duration=1s. В очереди: 1
2025-05-09 20:02:20,812 - TimeTracker - INFO - Started new activity session: App='python.exe', Title='Выход, Useful='None', StartUTC='2025-05-09T17:02:20.812114Z'
2025-05-09 20:02:21,498 - TimeTracker - INFO - Завершена сессия активности: App='python.exe', Title='Выход, Duration=1s. В очереди: 2
2025-05-09 20:13:37,349 - TimeTracker - INFO - Конфигурация успешно загружена из C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 20:13:37,349 - TimeTracker - INFO - Загружено 2 отслеживаемых приложений.
2025-05-09 20:14:52,998 - TimeTracker - INFO - Конфигурация успешно загружена из C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 20:14:52,999 - TimeTracker - INFO - Загружено 2 отслеживаемых приложений.
2025-05-09 20:15:50,361 - TimeTracker - INFO - Конфигурация успешно загружена из C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 20:15:50,361 - TimeTracker - INFO - Загружено 2 отслеживаемых приложений.
2025-05-09 20:15:50,362 - TimeTracker - INFO - Конфигурация успешно сохранена в C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 20:16:12,028 - TimeTracker - INFO - Конфигурация успешно загружена из C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 20:16:12,029 - TimeTracker - INFO - Загружено 2 отслеживаемых приложений.
2025-05-09 20:16:12,029 - TimeTracker - INFO - Конфигурация успешно сохранена в C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 20:35:21,258 - TimeTracker - INFO - Конфигурация успешно загружена из C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 20:35:21,259 - TimeTracker - INFO - Загружено 2 отслеживаемых приложений.
2025-05-09 21:01:05,207 - TimeTracker - INFO - Конфигурация успешно загружена из C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 21:01:05,208 - TimeTracker - INFO - Загружено 2 отслеживаемых приложений.
2025-05-09 21:01:05,209 - TimeTracker - INFO - Конфигурация успешно сохранена в C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 21:01:05,794 - TimeTracker - INFO - Обновлен список приложений: 106 процессов
2025-05-09 21:01:05,818 - TimeTracker - INFO - Запущен поток отслеживания активности
2025-05-09 21:01:06,932 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:01:10,539 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:01:11,940 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:01:15,548 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:01:16,945 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:01:20,520 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:01:21,951 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:01:25,550 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:01:26,351 - TimeTracker - INFO - Конфигурация успешно сохранена в C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 21:01:26,757 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:01:26,757 - TimeTracker - INFO - Изменен статус полезности приложения 'winlogon.exe (Не отслеживается, Запущено)': True
2025-05-09 21:01:26,958 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:01:30,584 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:01:31,965 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:01:35,212 - TimeTracker - WARNING - Токен истек 2025-05-09 20:41:26. Требуется повторная авторизация.
2025-05-09 21:01:35,632 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:01:36,213 - TimeTracker - WARNING - Токен истек 2025-05-09 20:41:26. Требуется повторная авторизация.
2025-05-09 21:01:36,213 - TimeTracker - INFO - Токен отсутствует или недействителен, вызывается диалог входа.
2025-05-09 21:01:36,971 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:01:40,540 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:01:41,977 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:01:45,112 - TimeTracker - INFO - Попытка авторизации на сервере http://127.0.0.1:8000 с логином Heist
2025-05-09 21:01:45,112 - TimeTracker - INFO - Авторизация на URL: http://127.0.0.1:8000/api/token/
2025-05-09 21:01:45,717 - TimeTracker - INFO - Успешная авторизация на сервере
2025-05-09 21:01:45,718 - TimeTracker - INFO - Получение списка приложений с URL: http://127.0.0.1:8000/api/applications/
2025-05-09 21:01:45,724 - TimeTracker - INFO - Загружено 7 приложений с сервера
2025-05-09 21:01:45,725 - TimeTracker - INFO - Конфигурация успешно сохранена в C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 21:01:45,725 - TimeTracker - INFO - Токен авторизации успешно сохранен в конфигурации.
2025-05-09 21:01:45,729 - TimeTracker - INFO - Вход выполнен успешно через диалог.
2025-05-09 21:01:46,117 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:01:46,984 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:01:49,328 - TimeTracker - INFO - Открываем веб-интерфейс: http://127.0.0.1:8000/dashboard/
2025-05-09 21:01:50,822 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:01:51,990 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:01:55,783 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:01:56,997 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:02:00,817 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:02:02,004 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:02:05,217 - TimeTracker - INFO - Токен действителен еще 1.0 часов
2025-05-09 21:02:05,653 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:02:07,012 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:02:10,559 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:02:12,019 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:02:15,552 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:02:17,025 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:02:20,607 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:02:22,032 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:02:25,621 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:02:27,038 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:02:30,569 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:02:32,045 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:02:35,210 - TimeTracker - INFO - Токен действителен еще 1.0 часов
2025-05-09 21:02:35,611 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:02:37,051 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:02:40,576 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:02:42,058 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:02:45,570 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:02:47,064 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:02:50,647 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:02:52,070 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:02:55,813 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:02:57,077 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:02:58,391 - TimeTracker - INFO - Отслеживание активности запущено
2025-05-09 21:03:00,578 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:03:02,084 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:03:05,204 - TimeTracker - INFO - Токен действителен еще 1.0 часов
2025-05-09 21:03:05,564 - TimeTracker - INFO - Обновлен список приложений: 105 процессов
2025-05-09 21:03:07,090 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:03:10,537 - TimeTracker - INFO - Обновлен список приложений: 105 процессов
2025-05-09 21:03:12,096 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:03:15,554 - TimeTracker - INFO - Обновлен список приложений: 105 процессов
2025-05-09 21:03:17,102 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:03:20,589 - TimeTracker - INFO - Обновлен список приложений: 105 процессов
2025-05-09 21:03:22,109 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:03:25,558 - TimeTracker - INFO - Обновлен список приложений: 105 процессов
2025-05-09 21:03:27,115 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:03:30,579 - TimeTracker - INFO - Обновлен список приложений: 105 процессов
2025-05-09 21:03:32,122 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:03:35,216 - TimeTracker - INFO - Токен действителен еще 1.0 часов
2025-05-09 21:03:35,572 - TimeTracker - INFO - Обновлен список приложений: 105 процессов
2025-05-09 21:03:37,128 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:03:40,565 - TimeTracker - INFO - Обновлен список приложений: 105 процессов
2025-05-09 21:03:42,135 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:03:45,572 - TimeTracker - INFO - Обновлен список приложений: 105 процессов
2025-05-09 21:03:47,142 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:03:50,818 - TimeTracker - INFO - Обновлен список приложений: 105 процессов
2025-05-09 21:03:52,148 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 640, in track_activity
    active_window_info = self.get_active_window_info()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'get_active_window_info'
2025-05-09 21:05:15,873 - TimeTracker - INFO - Конфигурация успешно загружена из C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 21:05:15,873 - TimeTracker - INFO - Загружено 3 отслеживаемых приложений.
2025-05-09 21:05:15,875 - TimeTracker - INFO - Конфигурация успешно сохранена в C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 21:05:16,632 - TimeTracker - INFO - Обновлен список приложений: 106 процессов
2025-05-09 21:05:16,698 - TimeTracker - INFO - Запущен поток отслеживания активности
2025-05-09 21:05:17,699 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:05:21,250 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:05:22,706 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:05:26,319 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:05:27,713 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:05:31,282 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:05:32,719 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:05:34,451 - TimeTracker - INFO - Открываем веб-интерфейс: http://127.0.0.1:8000/dashboard/
2025-05-09 21:05:36,321 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:05:37,727 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:05:41,353 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:05:42,735 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:05:45,869 - TimeTracker - INFO - Токен действителен еще 0.9 часов
2025-05-09 21:05:46,537 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:05:47,741 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:05:51,374 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:05:52,752 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:05:56,291 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:05:57,758 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:06:01,281 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:06:02,765 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:06:06,301 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:06:07,772 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:06:11,290 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:06:12,779 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:06:15,865 - TimeTracker - INFO - Токен действителен еще 0.9 часов
2025-05-09 21:06:16,285 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:06:17,786 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:06:21,275 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:06:22,792 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:06:26,290 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:06:27,800 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:06:31,263 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:06:32,806 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:06:36,302 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:06:37,314 - TimeTracker - INFO - Конфигурация успешно сохранена в C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 21:06:37,736 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:06:37,736 - TimeTracker - INFO - Изменен статус отслеживания приложения 'wallpaper64.exe (Не отслеживается, Запущено)': True
2025-05-09 21:06:37,813 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:06:40,594 - TimeTracker - INFO - Конфигурация успешно сохранена в C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 21:06:41,064 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:06:41,064 - TimeTracker - INFO - Изменен статус отслеживания приложения 'wallpaperservice32.exe (Не отслеживается, Запущено)': True
2025-05-09 21:06:41,473 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:06:42,820 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:06:45,861 - TimeTracker - INFO - Токен действителен еще 0.9 часов
2025-05-09 21:06:46,317 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:06:47,827 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:06:49,993 - TimeTracker - INFO - Изменения в SettingsDialog отменены.
2025-05-09 21:06:51,357 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:06:52,833 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:06:56,418 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:06:57,841 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:07:01,455 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:07:02,848 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:07:06,377 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:07:07,855 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:07:11,383 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:07:12,862 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:07:15,861 - TimeTracker - INFO - Токен действителен еще 0.9 часов
2025-05-09 21:07:16,467 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:07:17,868 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:07:21,271 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:07:22,875 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:07:26,336 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:07:27,881 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:07:31,329 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:07:32,888 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:07:36,464 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:07:37,897 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:07:41,495 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:07:42,903 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:07:45,861 - TimeTracker - INFO - Токен действителен еще 0.9 часов
2025-05-09 21:07:46,526 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:07:47,911 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:07:51,517 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:07:52,918 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:07:56,476 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:07:57,925 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:08:01,468 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:08:02,931 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:08:06,371 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:08:07,938 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:08:09,973 - TimeTracker - INFO - Конфигурация успешно сохранена в C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 21:08:10,370 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:08:10,370 - TimeTracker - INFO - Изменен статус отслеживания приложения 'webwallpaper32.exe (Не отслеживается, Запущено)': True
2025-05-09 21:08:11,315 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:08:12,945 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:08:15,869 - TimeTracker - INFO - Токен действителен еще 0.9 часов
2025-05-09 21:08:16,348 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:08:17,952 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:08:21,341 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:08:22,959 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:08:26,361 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:08:27,965 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:08:31,375 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:08:32,972 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:08:36,377 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:08:37,507 - TimeTracker - INFO - Открываем веб-интерфейс: http://127.0.0.1:8000/dashboard/
2025-05-09 21:08:37,979 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:08:41,452 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:08:42,986 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:08:45,864 - TimeTracker - INFO - Токен действителен еще 0.9 часов
2025-05-09 21:08:46,661 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:08:47,995 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:08:51,453 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:08:53,003 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:08:56,366 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:08:56,376 - TimeTracker - INFO - Конфигурация успешно сохранена в C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 21:08:56,791 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:08:56,791 - TimeTracker - INFO - Изменен статус отслеживания приложения 'lghub_agent.exe (Не отслеживается, Запущено)': True
2025-05-09 21:08:58,010 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:09:00,479 - TimeTracker - INFO - Конфигурация успешно сохранена в C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 21:09:00,878 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:09:00,879 - TimeTracker - INFO - Изменен статус отслеживания приложения 'nissrv.exe (Не отслеживается, Запущено)': True
2025-05-09 21:09:01,380 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:09:03,017 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:09:06,391 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:09:08,023 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:09:11,382 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:09:13,030 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:09:15,861 - TimeTracker - INFO - Токен действителен еще 0.9 часов
2025-05-09 21:09:16,373 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:09:18,037 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:09:21,385 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:09:23,044 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:09:26,376 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:09:28,051 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:09:28,829 - TimeTracker - INFO - Изменения в SettingsDialog отменены.
2025-05-09 21:09:31,435 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:09:33,058 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:09:36,405 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:09:38,065 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:09:41,477 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:09:43,072 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:09:45,875 - TimeTracker - INFO - Токен действителен еще 0.9 часов
2025-05-09 21:09:46,448 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:09:48,079 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:09:51,453 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:09:53,087 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:09:56,465 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:09:58,094 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:10:01,394 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:10:03,100 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:10:06,482 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:10:08,109 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:10:11,418 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:10:13,117 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:10:15,866 - TimeTracker - INFO - Токен действителен еще 0.9 часов
2025-05-09 21:10:16,463 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:10:18,124 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:10:21,414 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:10:23,131 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:10:26,392 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:10:28,138 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:10:31,409 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:10:33,145 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:10:36,395 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:10:38,152 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:10:41,438 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:10:43,160 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:10:45,869 - TimeTracker - INFO - Токен действителен еще 0.8 часов
2025-05-09 21:10:46,403 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:10:48,167 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:10:51,445 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:10:53,175 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:10:56,398 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:10:58,181 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:11:01,406 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:11:03,188 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:11:06,403 - TimeTracker - INFO - Обновлен список приложений: 106 процессов
2025-05-09 21:11:08,195 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:11:11,396 - TimeTracker - INFO - Обновлен список приложений: 105 процессов
2025-05-09 21:11:13,202 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:11:15,871 - TimeTracker - INFO - Токен действителен еще 0.8 часов
2025-05-09 21:11:16,397 - TimeTracker - INFO - Обновлен список приложений: 105 процессов
2025-05-09 21:11:18,209 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:11:21,389 - TimeTracker - INFO - Обновлен список приложений: 105 процессов
2025-05-09 21:11:23,216 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:11:26,408 - TimeTracker - INFO - Обновлен список приложений: 105 процессов
2025-05-09 21:11:28,223 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:11:31,420 - TimeTracker - INFO - Обновлен список приложений: 105 процессов
2025-05-09 21:11:33,230 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:11:36,402 - TimeTracker - INFO - Обновлен список приложений: 106 процессов
2025-05-09 21:11:38,237 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:15:01,578 - TimeTracker - INFO - Конфигурация успешно загружена из C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 21:15:01,579 - TimeTracker - INFO - Загружено 8 отслеживаемых приложений.
2025-05-09 21:15:01,579 - TimeTracker - INFO - Конфигурация успешно сохранена в C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 21:15:02,219 - TimeTracker - INFO - Обновлен список приложений: 106 процессов
2025-05-09 21:15:02,245 - TimeTracker - INFO - Запущен поток отслеживания активности
2025-05-09 21:15:03,358 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:15:08,365 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:15:11,972 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:15:13,371 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:15:18,378 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:15:21,951 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:15:23,384 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:15:28,392 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:15:31,566 - TimeTracker - INFO - Токен действителен еще 0.8 часов
2025-05-09 21:15:31,946 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:15:33,399 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:15:38,406 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:15:42,013 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:15:43,413 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:15:48,420 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:15:51,950 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:15:53,427 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:15:58,434 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:16:00,099 - TimeTracker - INFO - Конфигурация успешно сохранена в C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 21:16:00,463 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:16:01,578 - TimeTracker - INFO - Токен действителен еще 0.8 часов
2025-05-09 21:16:01,900 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:16:03,441 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:16:04,933 - TimeTracker - INFO - Изменен статус отслеживания приложения 'chatgpt.exe (Не отслеживается, Запущено)': включено
2025-05-09 21:16:08,447 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:16:08,474 - TimeTracker - INFO - Конфигурация успешно сохранена в C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 21:16:08,815 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:16:11,937 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:16:12,913 - TimeTracker - INFO - Изменен статус отслеживания приложения 'chatgpt.exe (Полезное, Запущено)': отключено
2025-05-09 21:16:13,454 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:16:18,461 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:16:21,525 - TimeTracker - INFO - Открываем веб-интерфейс: http://127.0.0.1:8000/dashboard/
2025-05-09 21:16:22,297 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:16:23,468 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:16:28,475 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:16:31,575 - TimeTracker - INFO - Токен действителен еще 0.8 часов
2025-05-09 21:16:31,929 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:16:33,482 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:16:37,169 - TimeTracker - INFO - Конфигурация успешно сохранена в C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 21:16:37,504 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:16:38,492 - TimeTracker - INFO - Изменен статус отслеживания приложения 'cursor.exe (Не отслеживается, Запущено)': включено
2025-05-09 21:16:38,492 - TimeTracker - ERROR - Ошибка при получении информации об активном окне: pid must be a positive integer (got -137906688)
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 1386, in get_active_window_info
    process = psutil.Process(process_id)
  File "C:\Users\mihai\AppData\Roaming\Python\Python313\site-packages\psutil\__init__.py", line 332, in __init__
    self._init(pid)
    ~~~~~~~~~~^^^^^
  File "C:\Users\mihai\AppData\Roaming\Python\Python313\site-packages\psutil\__init__.py", line 341, in _init
    raise ValueError('pid must be a positive integer (got %s)'
                     % pid)
ValueError: pid must be a positive integer (got -137906688)
2025-05-09 21:16:39,494 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:16:41,979 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:16:44,501 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:16:49,508 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:16:51,986 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:16:54,515 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:16:55,625 - TimeTracker - INFO - Отслеживание активности запущено
2025-05-09 21:16:59,522 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:17:01,569 - TimeTracker - INFO - Токен действителен еще 0.7 часов
2025-05-09 21:17:02,290 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:17:04,529 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:17:09,537 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:17:11,955 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:17:14,545 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:17:19,552 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:17:22,266 - TimeTracker - INFO - Обновлен список приложений: 109 процессов
2025-05-09 21:17:24,558 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:17:29,566 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:17:31,568 - TimeTracker - INFO - Токен действителен еще 0.7 часов
2025-05-09 21:17:31,994 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:17:34,574 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:17:39,581 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:17:41,945 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:17:44,587 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:17:49,595 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:17:51,939 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:17:54,601 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:17:59,611 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:18:01,575 - TimeTracker - INFO - Токен действителен еще 0.7 часов
2025-05-09 21:18:01,935 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:18:04,618 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:18:09,625 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:18:11,964 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:18:14,634 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:18:19,641 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:18:21,935 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:18:24,647 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:18:29,654 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:18:31,574 - TimeTracker - INFO - Токен действителен еще 0.7 часов
2025-05-09 21:18:31,927 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:18:35,662 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:18:40,672 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:18:41,901 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:18:45,679 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:18:50,685 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:18:51,937 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:18:55,692 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:19:00,698 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:19:01,570 - TimeTracker - INFO - Токен действителен еще 0.7 часов
2025-05-09 21:19:01,912 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:19:05,705 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:19:10,711 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:19:11,905 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:19:15,719 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:19:20,726 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:19:21,943 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:19:25,733 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:19:30,739 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:19:31,578 - TimeTracker - INFO - Токен действителен еще 0.7 часов
2025-05-09 21:19:32,026 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:19:35,747 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:19:40,753 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:19:41,954 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:19:45,760 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:19:50,768 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:19:52,080 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:19:55,774 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:20:00,781 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:20:01,574 - TimeTracker - INFO - Токен действителен еще 0.7 часов
2025-05-09 21:20:02,210 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:20:05,789 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:20:10,797 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:20:12,310 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:20:15,804 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:20:20,812 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:20:22,294 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:20:25,819 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:20:30,827 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:20:31,565 - TimeTracker - INFO - Токен действителен еще 0.7 часов
2025-05-09 21:20:32,371 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:20:35,838 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:20:40,850 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:20:42,182 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:20:45,858 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:20:50,869 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:20:52,150 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:20:55,876 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:21:00,883 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:21:01,572 - TimeTracker - INFO - Токен действителен еще 0.7 часов
2025-05-09 21:21:02,228 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:21:05,891 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:21:10,898 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:21:12,081 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:21:15,905 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:21:20,913 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:21:22,018 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:21:25,919 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:21:30,926 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:21:31,573 - TimeTracker - INFO - Токен действителен еще 0.7 часов
2025-05-09 21:21:32,087 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:21:35,933 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:21:40,940 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:21:42,115 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:21:45,948 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:21:50,955 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:21:52,077 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:21:55,962 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:22:00,969 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:22:01,567 - TimeTracker - INFO - Токен действителен еще 0.7 часов
2025-05-09 21:22:02,093 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:22:05,976 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:22:10,983 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:22:12,134 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:22:15,989 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:22:20,998 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:22:21,997 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:22:26,005 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    is_useful = self.is_app_useful(app_name)
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:22:31,012 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:22:31,570 - TimeTracker - INFO - Токен действителен еще 0.7 часов
2025-05-09 21:22:32,097 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:22:36,019 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:22:41,027 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:22:42,064 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:22:46,034 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:22:51,041 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:22:52,076 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:22:56,048 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:23:01,056 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:23:01,566 - TimeTracker - INFO - Токен действителен еще 0.6 часов
2025-05-09 21:23:02,189 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:23:06,066 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:23:11,073 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:23:12,105 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:23:16,081 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:23:21,089 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:23:21,990 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:23:26,095 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:23:31,102 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:23:31,571 - TimeTracker - INFO - Токен действителен еще 0.6 часов
2025-05-09 21:23:32,009 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:23:36,108 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:23:41,115 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:23:41,926 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:23:46,122 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:23:49,449 - TimeTracker - INFO - Конфигурация успешно загружена из C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 21:23:49,450 - TimeTracker - INFO - Загружено 10 отслеживаемых приложений.
2025-05-09 21:23:49,450 - TimeTracker - INFO - Конфигурация успешно сохранена в C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 21:23:50,088 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:23:50,112 - TimeTracker - INFO - Запущен поток отслеживания активности
2025-05-09 21:23:51,129 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:23:51,221 - TimeTracker - INFO - Начата новая сессия активности: App='python.exe', Title='Time Tracker PRO'
2025-05-09 21:23:51,974 - TimeTracker - INFO - Обновлен список приложений: 108 процессов
2025-05-09 21:23:56,136 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:23:58,364 - TimeTracker - INFO - Открываем веб-интерфейс: http://127.0.0.1:8000/dashboard/
2025-05-09 21:23:59,227 - TimeTracker - INFO - Добавлено 1 нажатий клавиш в активность
2025-05-09 21:23:59,228 - TimeTracker - INFO - Завершена сессия активности: App='python.exe', Title='Time Tracker PRO, Duration=8s. В очереди: 1
2025-05-09 21:23:59,229 - TimeTracker - INFO - Начата новая сессия активности: App='chrome.exe', Title='Главная - Time Tracker - Googl...'
2025-05-09 21:23:59,453 - TimeTracker - INFO - Вычисленная длительность: 8.009071 секунд
2025-05-09 21:23:59,453 - TimeTracker - INFO - Отправляем запрос на создание приложения: http://127.0.0.1:8000/api/applications/
2025-05-09 21:23:59,477 - TimeTracker - INFO - Создано новое приложение: python.exe с ID=21
2025-05-09 21:23:59,477 - TimeTracker - INFO - Для приложения python.exe выбран ID=21
2025-05-09 21:23:59,477 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:23:51.219051Z, end_time=2025-05-09T18:23:59.228122Z, длительность=8.009071 секунд
2025-05-09 21:23:59,478 - TimeTracker - INFO - Отправка 1 записей активности на сервер.
2025-05-09 21:23:59,478 - TimeTracker - INFO - API URL: http://127.0.0.1:8000/api/activities/, Токен: eyJhbGciOi...
2025-05-09 21:23:59,478 - TimeTracker - INFO - Отправляем пайлоад: {'application': 21, 'title': 'Time Tracker PRO', 'start_time': '2025-05-09T18:23:51.219051Z', 'end_time': '2025-05-09T18:23:59.228122Z', 'is_productive': False, 'app_name': 'python.exe', 'keyboard_presses': 1}
2025-05-09 21:23:59,491 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":325,"user":3,"application":21,"start_time":"2025-05-09T21:23:51.219051+03:00","end_time":"2025-05-09T21:23:59.228122+03:00","duration":"00:00:08.009071","keyboard_presses":1}
2025-05-09 21:23:59,491 - TimeTracker - INFO - Успешно отправлено 1 записей активности.
2025-05-09 21:23:59,987 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:24:01,144 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:24:01,568 - TimeTracker - INFO - Токен действителен еще 0.6 часов
2025-05-09 21:24:02,009 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:24:03,231 - TimeTracker - INFO - Добавлено 0 нажатий клавиш в активность
2025-05-09 21:24:03,231 - TimeTracker - INFO - Завершена сессия активности: App='chrome.exe', Title='Главная - Time Tracker - Googl..., Duration=4s. В очереди: 1
2025-05-09 21:24:03,233 - TimeTracker - INFO - Начата новая сессия активности: App='chrome.exe', Title='Статистика - Time Tracker - Go...'
2025-05-09 21:24:05,235 - TimeTracker - INFO - Добавлено 0 нажатий клавиш в активность
2025-05-09 21:24:05,235 - TimeTracker - INFO - Завершена сессия активности: App='chrome.exe', Title='Статистика - Time Tracker - Go..., Duration=2s. В очереди: 2
2025-05-09 21:24:05,237 - TimeTracker - INFO - Начата новая сессия активности: App='chrome.exe', Title='Логи - Time Tracker - Google C...'
2025-05-09 21:24:06,154 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:24:06,238 - TimeTracker - INFO - Добавлено 0 нажатий клавиш в активность
2025-05-09 21:24:06,239 - TimeTracker - INFO - Завершена сессия активности: App='chrome.exe', Title='Логи - Time Tracker - Google C..., Duration=1s. В очереди: 3
2025-05-09 21:24:06,242 - TimeTracker - INFO - Начата новая сессия активности: App='chrome.exe', Title='Tracker33 - Google Chrome'
2025-05-09 21:24:07,243 - TimeTracker - INFO - Добавлено 0 нажатий клавиш в активность
2025-05-09 21:24:07,244 - TimeTracker - INFO - Завершена сессия активности: App='chrome.exe', Title='Tracker33 - Google Chrome, Duration=1s. В очереди: 4
2025-05-09 21:24:07,246 - TimeTracker - INFO - Начата новая сессия активности: App='chrome.exe', Title='Логи - Time Tracker - Google C...'
2025-05-09 21:24:08,249 - TimeTracker - INFO - Добавлено 0 нажатий клавиш в активность
2025-05-09 21:24:08,249 - TimeTracker - INFO - Завершена сессия активности: App='chrome.exe', Title='Логи - Time Tracker - Google C..., Duration=1s. В очереди: 5
2025-05-09 21:24:08,253 - TimeTracker - INFO - Начата новая сессия активности: App='chrome.exe', Title='Главная - Time Tracker - Googl...'
2025-05-09 21:24:09,439 - TimeTracker - INFO - Вычисленная длительность: 4.002613 секунд
2025-05-09 21:24:09,439 - TimeTracker - INFO - Отправляем запрос на создание приложения: http://127.0.0.1:8000/api/applications/
2025-05-09 21:24:09,449 - TimeTracker - INFO - Создано новое приложение: chrome.exe с ID=22
2025-05-09 21:24:09,449 - TimeTracker - INFO - Для приложения chrome.exe выбран ID=22
2025-05-09 21:24:09,450 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:23:59.228895Z, end_time=2025-05-09T18:24:03.231508Z, длительность=4.002613 секунд
2025-05-09 21:24:09,450 - TimeTracker - INFO - Вычисленная длительность: 2.002599 секунд
2025-05-09 21:24:09,450 - TimeTracker - INFO - Для приложения chrome.exe выбран ID=22
2025-05-09 21:24:09,450 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:24:03.232916Z, end_time=2025-05-09T18:24:05.235515Z, длительность=2.002599 секунд
2025-05-09 21:24:09,450 - TimeTracker - INFO - Вычисленная длительность: 1.00231 секунд
2025-05-09 21:24:09,450 - TimeTracker - INFO - Для приложения chrome.exe выбран ID=22
2025-05-09 21:24:09,451 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:24:05.236727Z, end_time=2025-05-09T18:24:06.239037Z, длительность=1.00231 секунд
2025-05-09 21:24:09,451 - TimeTracker - INFO - Вычисленная длительность: 1.003364 секунд
2025-05-09 21:24:09,451 - TimeTracker - INFO - Для приложения chrome.exe выбран ID=22
2025-05-09 21:24:09,451 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:24:06.240876Z, end_time=2025-05-09T18:24:07.244240Z, длительность=1.003364 секунд
2025-05-09 21:24:09,451 - TimeTracker - INFO - Вычисленная длительность: 1.003806 секунд
2025-05-09 21:24:09,451 - TimeTracker - INFO - Для приложения chrome.exe выбран ID=22
2025-05-09 21:24:09,452 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:24:07.245897Z, end_time=2025-05-09T18:24:08.249703Z, длительность=1.003806 секунд
2025-05-09 21:24:09,452 - TimeTracker - INFO - Отправка 5 записей активности на сервер.
2025-05-09 21:24:09,452 - TimeTracker - INFO - API URL: http://127.0.0.1:8000/api/activities/, Токен: eyJhbGciOi...
2025-05-09 21:24:09,452 - TimeTracker - INFO - Отправляем пайлоад: {'application': 22, 'title': 'Главная - Time Tracker - Google Chrome', 'start_time': '2025-05-09T18:23:59.228895Z', 'end_time': '2025-05-09T18:24:03.231508Z', 'is_productive': False, 'app_name': 'chrome.exe', 'keyboard_presses': 0}
2025-05-09 21:24:09,462 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":326,"user":3,"application":22,"start_time":"2025-05-09T21:23:59.228895+03:00","end_time":"2025-05-09T21:24:03.231508+03:00","duration":"00:00:04.002613","keyboard_presses":0}
2025-05-09 21:24:09,463 - TimeTracker - INFO - Отправляем пайлоад: {'application': 22, 'title': 'Статистика - Time Tracker - Google Chrome', 'start_time': '2025-05-09T18:24:03.232916Z', 'end_time': '2025-05-09T18:24:05.235515Z', 'is_productive': False, 'app_name': 'chrome.exe', 'keyboard_presses': 0}
2025-05-09 21:24:09,473 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":327,"user":3,"application":22,"start_time":"2025-05-09T21:24:03.232916+03:00","end_time":"2025-05-09T21:24:05.235515+03:00","duration":"00:00:02.002599","keyboard_presses":0}
2025-05-09 21:24:09,473 - TimeTracker - INFO - Отправляем пайлоад: {'application': 22, 'title': 'Логи - Time Tracker - Google Chrome', 'start_time': '2025-05-09T18:24:05.236727Z', 'end_time': '2025-05-09T18:24:06.239037Z', 'is_productive': False, 'app_name': 'chrome.exe', 'keyboard_presses': 0}
2025-05-09 21:24:09,483 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":328,"user":3,"application":22,"start_time":"2025-05-09T21:24:05.236727+03:00","end_time":"2025-05-09T21:24:06.239037+03:00","duration":"00:00:01.002310","keyboard_presses":0}
2025-05-09 21:24:09,483 - TimeTracker - INFO - Отправляем пайлоад: {'application': 22, 'title': 'Tracker33 - Google Chrome', 'start_time': '2025-05-09T18:24:06.240876Z', 'end_time': '2025-05-09T18:24:07.244240Z', 'is_productive': False, 'app_name': 'chrome.exe', 'keyboard_presses': 0}
2025-05-09 21:24:09,494 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":329,"user":3,"application":22,"start_time":"2025-05-09T21:24:06.240876+03:00","end_time":"2025-05-09T21:24:07.244240+03:00","duration":"00:00:01.003364","keyboard_presses":0}
2025-05-09 21:24:09,494 - TimeTracker - INFO - Отправляем пайлоад: {'application': 22, 'title': 'Логи - Time Tracker - Google Chrome', 'start_time': '2025-05-09T18:24:07.245897Z', 'end_time': '2025-05-09T18:24:08.249703Z', 'is_productive': False, 'app_name': 'chrome.exe', 'keyboard_presses': 0}
2025-05-09 21:24:09,505 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":330,"user":3,"application":22,"start_time":"2025-05-09T21:24:07.245897+03:00","end_time":"2025-05-09T21:24:08.249703+03:00","duration":"00:00:01.003806","keyboard_presses":0}
2025-05-09 21:24:09,506 - TimeTracker - INFO - Успешно отправлено 5 записей активности.
2025-05-09 21:24:09,982 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:24:11,162 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:24:12,257 - TimeTracker - INFO - Добавлено 0 нажатий клавиш в активность
2025-05-09 21:24:12,257 - TimeTracker - INFO - Завершена сессия активности: App='chrome.exe', Title='Главная - Time Tracker - Googl..., Duration=4s. В очереди: 1
2025-05-09 21:24:12,262 - TimeTracker - INFO - Начата новая сессия активности: App='chrome.exe', Title='Логи - Time Tracker - Google C...'
2025-05-09 21:24:12,306 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:24:16,171 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:24:16,265 - TimeTracker - INFO - Добавлено 2 нажатий клавиш в активность
2025-05-09 21:24:16,265 - TimeTracker - INFO - Завершена сессия активности: App='chrome.exe', Title='Логи - Time Tracker - Google C..., Duration=4s. В очереди: 2
2025-05-09 21:24:16,268 - TimeTracker - INFO - Начата новая сессия активности: App='chrome.exe', Title='Статистика - Time Tracker - Go...'
2025-05-09 21:24:18,271 - TimeTracker - INFO - Добавлено 4 нажатий клавиш в активность
2025-05-09 21:24:18,271 - TimeTracker - INFO - Завершена сессия активности: App='chrome.exe', Title='Статистика - Time Tracker - Go..., Duration=2s. В очереди: 3
2025-05-09 21:24:19,273 - TimeTracker - INFO - Начата новая сессия активности: App='python.exe', Title='Time Tracker PRO'
2025-05-09 21:24:19,450 - TimeTracker - INFO - Токен действителен еще 0.6 часов
2025-05-09 21:24:19,451 - TimeTracker - INFO - Вычисленная длительность: 4.005761 секунд
2025-05-09 21:24:19,451 - TimeTracker - INFO - Для приложения chrome.exe выбран ID=22
2025-05-09 21:24:19,451 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:24:08.251637Z, end_time=2025-05-09T18:24:12.257398Z, длительность=4.005761 секунд
2025-05-09 21:24:19,451 - TimeTracker - INFO - Вычисленная длительность: 4.005113 секунд
2025-05-09 21:24:19,451 - TimeTracker - INFO - Для приложения chrome.exe выбран ID=22
2025-05-09 21:24:19,451 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:24:12.260788Z, end_time=2025-05-09T18:24:16.265901Z, длительность=4.005113 секунд
2025-05-09 21:24:19,451 - TimeTracker - INFO - Вычисленная длительность: 2.003594 секунд
2025-05-09 21:24:19,452 - TimeTracker - INFO - Для приложения chrome.exe выбран ID=22
2025-05-09 21:24:19,452 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:24:16.267858Z, end_time=2025-05-09T18:24:18.271452Z, длительность=2.003594 секунд
2025-05-09 21:24:19,452 - TimeTracker - INFO - Отправка 3 записей активности на сервер.
2025-05-09 21:24:19,452 - TimeTracker - INFO - API URL: http://127.0.0.1:8000/api/activities/, Токен: eyJhbGciOi...
2025-05-09 21:24:19,452 - TimeTracker - INFO - Отправляем пайлоад: {'application': 22, 'title': 'Главная - Time Tracker - Google Chrome', 'start_time': '2025-05-09T18:24:08.251637Z', 'end_time': '2025-05-09T18:24:12.257398Z', 'is_productive': False, 'app_name': 'chrome.exe', 'keyboard_presses': 0}
2025-05-09 21:24:19,462 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":331,"user":3,"application":22,"start_time":"2025-05-09T21:24:08.251637+03:00","end_time":"2025-05-09T21:24:12.257398+03:00","duration":"00:00:04.005761","keyboard_presses":0}
2025-05-09 21:24:19,462 - TimeTracker - INFO - Отправляем пайлоад: {'application': 22, 'title': 'Логи - Time Tracker - Google Chrome', 'start_time': '2025-05-09T18:24:12.260788Z', 'end_time': '2025-05-09T18:24:16.265901Z', 'is_productive': False, 'app_name': 'chrome.exe', 'keyboard_presses': 2}
2025-05-09 21:24:19,472 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":332,"user":3,"application":22,"start_time":"2025-05-09T21:24:12.260788+03:00","end_time":"2025-05-09T21:24:16.265901+03:00","duration":"00:00:04.005113","keyboard_presses":2}
2025-05-09 21:24:19,473 - TimeTracker - INFO - Отправляем пайлоад: {'application': 22, 'title': 'Статистика - Time Tracker - Google Chrome', 'start_time': '2025-05-09T18:24:16.267858Z', 'end_time': '2025-05-09T18:24:18.271452Z', 'is_productive': False, 'app_name': 'chrome.exe', 'keyboard_presses': 4}
2025-05-09 21:24:19,483 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":333,"user":3,"application":22,"start_time":"2025-05-09T21:24:16.267858+03:00","end_time":"2025-05-09T21:24:18.271452+03:00","duration":"00:00:02.003594","keyboard_presses":4}
2025-05-09 21:24:19,484 - TimeTracker - INFO - Успешно отправлено 3 записей активности.
2025-05-09 21:24:19,867 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:24:21,178 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:24:21,964 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:24:23,514 - TimeTracker - INFO - Отслеживание активности запущено
2025-05-09 21:24:26,186 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:24:29,813 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:24:31,198 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:24:31,570 - TimeTracker - INFO - Токен действителен еще 0.6 часов
2025-05-09 21:24:31,941 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:24:35,392 - TimeTracker - INFO - Добавлено 0 нажатий клавиш в активность
2025-05-09 21:24:35,392 - TimeTracker - INFO - Завершена сессия активности: App='python.exe', Title='Time Tracker PRO, Duration=16s. В очереди: 1
2025-05-09 21:24:35,394 - TimeTracker - INFO - Начата новая сессия активности: App='python.exe', Title='Настройки отслеживания'
2025-05-09 21:24:36,204 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:24:39,439 - TimeTracker - INFO - Вычисленная длительность: 16.119211 секунд
2025-05-09 21:24:39,440 - TimeTracker - INFO - Для приложения python.exe выбран ID=21
2025-05-09 21:24:39,440 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:24:19.273367Z, end_time=2025-05-09T18:24:35.392578Z, длительность=16.119211 секунд
2025-05-09 21:24:39,440 - TimeTracker - INFO - Отправка 1 записей активности на сервер.
2025-05-09 21:24:39,440 - TimeTracker - INFO - API URL: http://127.0.0.1:8000/api/activities/, Токен: eyJhbGciOi...
2025-05-09 21:24:39,440 - TimeTracker - INFO - Отправляем пайлоад: {'application': 21, 'title': 'Time Tracker PRO', 'start_time': '2025-05-09T18:24:19.273367Z', 'end_time': '2025-05-09T18:24:35.392578Z', 'is_productive': False, 'app_name': 'python.exe', 'keyboard_presses': 0}
2025-05-09 21:24:39,450 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":334,"user":3,"application":21,"start_time":"2025-05-09T21:24:19.273367+03:00","end_time":"2025-05-09T21:24:35.392578+03:00","duration":"00:00:16.119211","keyboard_presses":0}
2025-05-09 21:24:39,451 - TimeTracker - INFO - Успешно отправлено 1 записей активности.
2025-05-09 21:24:39,839 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:24:41,211 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:24:42,029 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:24:43,879 - TimeTracker - INFO - Сохранение настроек из SettingsDialog...
2025-05-09 21:24:43,880 - TimeTracker - INFO - Обновление конфигурации отслеживаемых приложений.
2025-05-09 21:24:43,880 - TimeTracker - INFO - Конфигурация успешно сохранена в C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 21:24:43,881 - TimeTracker - INFO - Конфигурация отслеживаемых приложений успешно обновлена и сохранена.
2025-05-09 21:24:44,399 - TimeTracker - INFO - Добавлено 0 нажатий клавиш в активность
2025-05-09 21:24:44,400 - TimeTracker - INFO - Завершена сессия активности: App='python.exe', Title='Настройки отслеживания, Duration=9s. В очереди: 1
2025-05-09 21:24:44,401 - TimeTracker - INFO - Начата новая сессия активности: App='python.exe', Title='Time Tracker PRO'
2025-05-09 21:24:46,218 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:24:49,438 - TimeTracker - INFO - Токен действителен еще 0.6 часов
2025-05-09 21:24:49,439 - TimeTracker - INFO - Вычисленная длительность: 9.006603 секунд
2025-05-09 21:24:49,439 - TimeTracker - INFO - Для приложения python.exe выбран ID=21
2025-05-09 21:24:49,439 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:24:35.393635Z, end_time=2025-05-09T18:24:44.400238Z, длительность=9.006603 секунд
2025-05-09 21:24:49,439 - TimeTracker - INFO - Отправка 1 записей активности на сервер.
2025-05-09 21:24:49,439 - TimeTracker - INFO - API URL: http://127.0.0.1:8000/api/activities/, Токен: eyJhbGciOi...
2025-05-09 21:24:49,440 - TimeTracker - INFO - Отправляем пайлоад: {'application': 21, 'title': 'Настройки отслеживания', 'start_time': '2025-05-09T18:24:35.393635Z', 'end_time': '2025-05-09T18:24:44.400238Z', 'is_productive': False, 'app_name': 'python.exe', 'keyboard_presses': 0}
2025-05-09 21:24:49,450 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":335,"user":3,"application":21,"start_time":"2025-05-09T21:24:35.393635+03:00","end_time":"2025-05-09T21:24:44.400238+03:00","duration":"00:00:09.006603","keyboard_presses":0}
2025-05-09 21:24:49,451 - TimeTracker - INFO - Успешно отправлено 1 записей активности.
2025-05-09 21:24:49,905 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:24:50,405 - TimeTracker - INFO - Добавлено 0 нажатий клавиш в активность
2025-05-09 21:24:50,405 - TimeTracker - INFO - Завершена сессия активности: App='python.exe', Title='Time Tracker PRO, Duration=6s. В очереди: 1
2025-05-09 21:24:50,407 - TimeTracker - INFO - Начата новая сессия активности: App='chrome.exe', Title='Статистика - Time Tracker - Go...'
2025-05-09 21:24:51,225 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:24:51,408 - TimeTracker - INFO - Добавлено 2 нажатий клавиш в активность
2025-05-09 21:24:51,409 - TimeTracker - INFO - Завершена сессия активности: App='chrome.exe', Title='Статистика - Time Tracker - Go..., Duration=1s. В очереди: 2
2025-05-09 21:24:51,410 - TimeTracker - INFO - Начата новая сессия активности: App='python.exe', Title='Time Tracker PRO'
2025-05-09 21:24:51,992 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:24:56,231 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:24:59,441 - TimeTracker - INFO - Вычисленная длительность: 6.004587 секунд
2025-05-09 21:24:59,441 - TimeTracker - INFO - Для приложения python.exe выбран ID=21
2025-05-09 21:24:59,441 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:24:44.400962Z, end_time=2025-05-09T18:24:50.405549Z, длительность=6.004587 секунд
2025-05-09 21:24:59,442 - TimeTracker - INFO - Вычисленная длительность: 1.002787 секунд
2025-05-09 21:24:59,442 - TimeTracker - INFO - Для приложения chrome.exe выбран ID=22
2025-05-09 21:24:59,442 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:24:50.406351Z, end_time=2025-05-09T18:24:51.409138Z, длительность=1.002787 секунд
2025-05-09 21:24:59,442 - TimeTracker - INFO - Отправка 2 записей активности на сервер.
2025-05-09 21:24:59,442 - TimeTracker - INFO - API URL: http://127.0.0.1:8000/api/activities/, Токен: eyJhbGciOi...
2025-05-09 21:24:59,442 - TimeTracker - INFO - Отправляем пайлоад: {'application': 21, 'title': 'Time Tracker PRO', 'start_time': '2025-05-09T18:24:44.400962Z', 'end_time': '2025-05-09T18:24:50.405549Z', 'is_productive': False, 'app_name': 'python.exe', 'keyboard_presses': 0}
2025-05-09 21:24:59,454 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":336,"user":3,"application":21,"start_time":"2025-05-09T21:24:44.400962+03:00","end_time":"2025-05-09T21:24:50.405549+03:00","duration":"00:00:06.004587","keyboard_presses":0}
2025-05-09 21:24:59,454 - TimeTracker - INFO - Отправляем пайлоад: {'application': 22, 'title': 'Статистика - Time Tracker - Google Chrome', 'start_time': '2025-05-09T18:24:50.406351Z', 'end_time': '2025-05-09T18:24:51.409138Z', 'is_productive': False, 'app_name': 'chrome.exe', 'keyboard_presses': 2}
2025-05-09 21:24:59,465 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":337,"user":3,"application":22,"start_time":"2025-05-09T21:24:50.406351+03:00","end_time":"2025-05-09T21:24:51.409138+03:00","duration":"00:00:01.002787","keyboard_presses":2}
2025-05-09 21:24:59,465 - TimeTracker - INFO - Успешно отправлено 2 записей активности.
2025-05-09 21:24:59,877 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:25:01,238 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:25:01,570 - TimeTracker - INFO - Токен действителен еще 0.6 часов
2025-05-09 21:25:01,915 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:25:05,417 - TimeTracker - INFO - Добавлено 0 нажатий клавиш в активность
2025-05-09 21:25:05,417 - TimeTracker - INFO - Завершена сессия активности: App='python.exe', Title='Time Tracker PRO, Duration=14s. В очереди: 1
2025-05-09 21:25:05,420 - TimeTracker - INFO - Начата новая сессия активности: App='chrome.exe', Title='Статистика - Time Tracker - Go...'
2025-05-09 21:25:06,245 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:25:06,421 - TimeTracker - INFO - Добавлено 2 нажатий клавиш в активность
2025-05-09 21:25:06,422 - TimeTracker - INFO - Завершена сессия активности: App='chrome.exe', Title='Статистика - Time Tracker - Go..., Duration=1s. В очереди: 2
2025-05-09 21:25:09,424 - TimeTracker - INFO - Начата новая сессия активности: App='python.exe', Title='Time Tracker PRO'
2025-05-09 21:25:09,453 - TimeTracker - INFO - Вычисленная длительность: 14.007769 секунд
2025-05-09 21:25:09,453 - TimeTracker - INFO - Для приложения python.exe выбран ID=21
2025-05-09 21:25:09,453 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:24:51.410082Z, end_time=2025-05-09T18:25:05.417851Z, длительность=14.007769 секунд
2025-05-09 21:25:09,454 - TimeTracker - INFO - Вычисленная длительность: 1.002872 секунд
2025-05-09 21:25:09,454 - TimeTracker - INFO - Для приложения chrome.exe выбран ID=22
2025-05-09 21:25:09,454 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:25:05.419226Z, end_time=2025-05-09T18:25:06.422098Z, длительность=1.002872 секунд
2025-05-09 21:25:09,454 - TimeTracker - INFO - Отправка 2 записей активности на сервер.
2025-05-09 21:25:09,454 - TimeTracker - INFO - API URL: http://127.0.0.1:8000/api/activities/, Токен: eyJhbGciOi...
2025-05-09 21:25:09,455 - TimeTracker - INFO - Отправляем пайлоад: {'application': 21, 'title': 'Time Tracker PRO', 'start_time': '2025-05-09T18:24:51.410082Z', 'end_time': '2025-05-09T18:25:05.417851Z', 'is_productive': False, 'app_name': 'python.exe', 'keyboard_presses': 0}
2025-05-09 21:25:09,465 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":338,"user":3,"application":21,"start_time":"2025-05-09T21:24:51.410082+03:00","end_time":"2025-05-09T21:25:05.417851+03:00","duration":"00:00:14.007769","keyboard_presses":0}
2025-05-09 21:25:09,465 - TimeTracker - INFO - Отправляем пайлоад: {'application': 22, 'title': 'Статистика - Time Tracker - Google Chrome', 'start_time': '2025-05-09T18:25:05.419226Z', 'end_time': '2025-05-09T18:25:06.422098Z', 'is_productive': False, 'app_name': 'chrome.exe', 'keyboard_presses': 2}
2025-05-09 21:25:09,475 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":339,"user":3,"application":22,"start_time":"2025-05-09T21:25:05.419226+03:00","end_time":"2025-05-09T21:25:06.422098+03:00","duration":"00:00:01.002872","keyboard_presses":2}
2025-05-09 21:25:09,475 - TimeTracker - INFO - Успешно отправлено 2 записей активности.
2025-05-09 21:25:09,837 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:25:10,426 - TimeTracker - INFO - Добавлено 2 нажатий клавиш в активность
2025-05-09 21:25:10,426 - TimeTracker - INFO - Завершена сессия активности: App='python.exe', Title='Time Tracker PRO, Duration=1s. В очереди: 1
2025-05-09 21:25:10,428 - TimeTracker - INFO - Начата новая сессия активности: App='chrome.exe', Title='Статистика - Time Tracker - Go...'
2025-05-09 21:25:11,253 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:25:11,429 - TimeTracker - INFO - Добавлено 2 нажатий клавиш в активность
2025-05-09 21:25:11,430 - TimeTracker - INFO - Завершена сессия активности: App='chrome.exe', Title='Статистика - Time Tracker - Go..., Duration=1s. В очереди: 2
2025-05-09 21:25:11,431 - TimeTracker - INFO - Начата новая сессия активности: App='python.exe', Title='Time Tracker PRO'
2025-05-09 21:25:11,962 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:25:12,432 - TimeTracker - INFO - Добавлено 2 нажатий клавиш в активность
2025-05-09 21:25:12,433 - TimeTracker - INFO - Завершена сессия активности: App='python.exe', Title='Time Tracker PRO, Duration=1s. В очереди: 3
2025-05-09 21:25:12,435 - TimeTracker - INFO - Начата новая сессия активности: App='chrome.exe', Title='Статистика - Time Tracker - Go...'
2025-05-09 21:25:13,436 - TimeTracker - INFO - Добавлено 2 нажатий клавиш в активность
2025-05-09 21:25:13,436 - TimeTracker - INFO - Завершена сессия активности: App='chrome.exe', Title='Статистика - Time Tracker - Go..., Duration=1s. В очереди: 4
2025-05-09 21:25:13,437 - TimeTracker - INFO - Начата новая сессия активности: App='python.exe', Title='Time Tracker PRO'
2025-05-09 21:25:16,263 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:25:19,451 - TimeTracker - INFO - Токен действителен еще 0.6 часов
2025-05-09 21:25:19,452 - TimeTracker - INFO - Вычисленная длительность: 1.002491 секунд
2025-05-09 21:25:19,452 - TimeTracker - INFO - Для приложения python.exe выбран ID=21
2025-05-09 21:25:19,452 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:25:09.424227Z, end_time=2025-05-09T18:25:10.426718Z, длительность=1.002491 секунд
2025-05-09 21:25:19,453 - TimeTracker - INFO - Вычисленная длительность: 1.002383 секунд
2025-05-09 21:25:19,453 - TimeTracker - INFO - Для приложения chrome.exe выбран ID=22
2025-05-09 21:25:19,453 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:25:10.427723Z, end_time=2025-05-09T18:25:11.430106Z, длительность=1.002383 секунд
2025-05-09 21:25:19,453 - TimeTracker - INFO - Вычисленная длительность: 1.002112 секунд
2025-05-09 21:25:19,453 - TimeTracker - INFO - Для приложения python.exe выбран ID=21
2025-05-09 21:25:19,453 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:25:11.430970Z, end_time=2025-05-09T18:25:12.433082Z, длительность=1.002112 секунд
2025-05-09 21:25:19,453 - TimeTracker - INFO - Вычисленная длительность: 1.001952 секунд
2025-05-09 21:25:19,454 - TimeTracker - INFO - Для приложения chrome.exe выбран ID=22
2025-05-09 21:25:19,454 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:25:12.434482Z, end_time=2025-05-09T18:25:13.436434Z, длительность=1.001952 секунд
2025-05-09 21:25:19,454 - TimeTracker - INFO - Отправка 4 записей активности на сервер.
2025-05-09 21:25:19,454 - TimeTracker - INFO - API URL: http://127.0.0.1:8000/api/activities/, Токен: eyJhbGciOi...
2025-05-09 21:25:19,454 - TimeTracker - INFO - Отправляем пайлоад: {'application': 21, 'title': 'Time Tracker PRO', 'start_time': '2025-05-09T18:25:09.424227Z', 'end_time': '2025-05-09T18:25:10.426718Z', 'is_productive': False, 'app_name': 'python.exe', 'keyboard_presses': 2}
2025-05-09 21:25:19,464 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":340,"user":3,"application":21,"start_time":"2025-05-09T21:25:09.424227+03:00","end_time":"2025-05-09T21:25:10.426718+03:00","duration":"00:00:01.002491","keyboard_presses":2}
2025-05-09 21:25:19,465 - TimeTracker - INFO - Отправляем пайлоад: {'application': 22, 'title': 'Статистика - Time Tracker - Google Chrome', 'start_time': '2025-05-09T18:25:10.427723Z', 'end_time': '2025-05-09T18:25:11.430106Z', 'is_productive': False, 'app_name': 'chrome.exe', 'keyboard_presses': 2}
2025-05-09 21:25:19,475 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":341,"user":3,"application":22,"start_time":"2025-05-09T21:25:10.427723+03:00","end_time":"2025-05-09T21:25:11.430106+03:00","duration":"00:00:01.002383","keyboard_presses":2}
2025-05-09 21:25:19,476 - TimeTracker - INFO - Отправляем пайлоад: {'application': 21, 'title': 'Time Tracker PRO', 'start_time': '2025-05-09T18:25:11.430970Z', 'end_time': '2025-05-09T18:25:12.433082Z', 'is_productive': False, 'app_name': 'python.exe', 'keyboard_presses': 2}
2025-05-09 21:25:19,485 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":342,"user":3,"application":21,"start_time":"2025-05-09T21:25:11.430970+03:00","end_time":"2025-05-09T21:25:12.433082+03:00","duration":"00:00:01.002112","keyboard_presses":2}
2025-05-09 21:25:19,486 - TimeTracker - INFO - Отправляем пайлоад: {'application': 22, 'title': 'Статистика - Time Tracker - Google Chrome', 'start_time': '2025-05-09T18:25:12.434482Z', 'end_time': '2025-05-09T18:25:13.436434Z', 'is_productive': False, 'app_name': 'chrome.exe', 'keyboard_presses': 2}
2025-05-09 21:25:19,495 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":343,"user":3,"application":22,"start_time":"2025-05-09T21:25:12.434482+03:00","end_time":"2025-05-09T21:25:13.436434+03:00","duration":"00:00:01.001952","keyboard_presses":2}
2025-05-09 21:25:19,495 - TimeTracker - INFO - Успешно отправлено 4 записей активности.
2025-05-09 21:25:19,908 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:25:21,270 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:25:22,052 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:25:23,443 - TimeTracker - INFO - Добавлено 0 нажатий клавиш в активность
2025-05-09 21:25:23,443 - TimeTracker - INFO - Завершена сессия активности: App='python.exe', Title='Time Tracker PRO, Duration=10s. В очереди: 1
2025-05-09 21:25:23,444 - TimeTracker - INFO - Приложение Cursor.exe находится в списке неотслеживаемых.
2025-05-09 21:25:26,279 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:25:27,446 - TimeTracker - INFO - Начата новая сессия активности: App='python.exe', Title='Time Tracker PRO'
2025-05-09 21:25:29,447 - TimeTracker - INFO - Вычисленная длительность: 10.006054 секунд
2025-05-09 21:25:29,448 - TimeTracker - INFO - Для приложения python.exe выбран ID=21
2025-05-09 21:25:29,448 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:25:13.437247Z, end_time=2025-05-09T18:25:23.443301Z, длительность=10.006054 секунд
2025-05-09 21:25:29,448 - TimeTracker - INFO - Отправка 1 записей активности на сервер.
2025-05-09 21:25:29,448 - TimeTracker - INFO - API URL: http://127.0.0.1:8000/api/activities/, Токен: eyJhbGciOi...
2025-05-09 21:25:29,448 - TimeTracker - INFO - Отправляем пайлоад: {'application': 21, 'title': 'Time Tracker PRO', 'start_time': '2025-05-09T18:25:13.437247Z', 'end_time': '2025-05-09T18:25:23.443301Z', 'is_productive': False, 'app_name': 'python.exe', 'keyboard_presses': 0}
2025-05-09 21:25:29,459 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":344,"user":3,"application":21,"start_time":"2025-05-09T21:25:13.437247+03:00","end_time":"2025-05-09T21:25:23.443301+03:00","duration":"00:00:10.006054","keyboard_presses":0}
2025-05-09 21:25:29,459 - TimeTracker - INFO - Успешно отправлено 1 записей активности.
2025-05-09 21:25:29,854 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:25:31,288 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:25:31,566 - TimeTracker - INFO - Токен действителен еще 0.6 часов
2025-05-09 21:25:31,953 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:25:36,295 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:25:39,842 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:25:39,842 - TimeTracker - INFO - Добавлено 0 нажатий клавиш в активность
2025-05-09 21:25:39,842 - TimeTracker - INFO - Завершена сессия активности: App='python.exe', Title='Time Tracker PRO, Duration=12s. В очереди: 1
2025-05-09 21:25:39,843 - TimeTracker - INFO - Начата новая сессия активности: App='python.exe', Title='Настройки отслеживания'
2025-05-09 21:25:41,302 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:25:41,956 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:25:46,309 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:25:46,909 - TimeTracker - INFO - Сохранение настроек из SettingsDialog...
2025-05-09 21:25:46,910 - TimeTracker - INFO - Обновление конфигурации отслеживаемых приложений.
2025-05-09 21:25:46,910 - TimeTracker - INFO - Конфигурация успешно сохранена в C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\config.ini
2025-05-09 21:25:46,911 - TimeTracker - INFO - Конфигурация отслеживаемых приложений успешно обновлена и сохранена.
2025-05-09 21:25:47,849 - TimeTracker - INFO - Добавлено 0 нажатий клавиш в активность
2025-05-09 21:25:47,849 - TimeTracker - INFO - Завершена сессия активности: App='python.exe', Title='Настройки отслеживания, Duration=8s. В очереди: 2
2025-05-09 21:25:47,850 - TimeTracker - INFO - Начата новая сессия активности: App='python.exe', Title='Time Tracker PRO'
2025-05-09 21:25:49,439 - TimeTracker - INFO - Токен действителен еще 0.6 часов
2025-05-09 21:25:49,440 - TimeTracker - INFO - Вычисленная длительность: 12.396364 секунд
2025-05-09 21:25:49,440 - TimeTracker - INFO - Для приложения python.exe выбран ID=21
2025-05-09 21:25:49,441 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:25:27.446417Z, end_time=2025-05-09T18:25:39.842781Z, длительность=12.396364 секунд
2025-05-09 21:25:49,441 - TimeTracker - INFO - Вычисленная длительность: 8.006006 секунд
2025-05-09 21:25:49,441 - TimeTracker - INFO - Для приложения python.exe выбран ID=21
2025-05-09 21:25:49,441 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:25:39.843498Z, end_time=2025-05-09T18:25:47.849504Z, длительность=8.006006 секунд
2025-05-09 21:25:49,441 - TimeTracker - INFO - Отправка 2 записей активности на сервер.
2025-05-09 21:25:49,442 - TimeTracker - INFO - API URL: http://127.0.0.1:8000/api/activities/, Токен: eyJhbGciOi...
2025-05-09 21:25:49,442 - TimeTracker - INFO - Отправляем пайлоад: {'application': 21, 'title': 'Time Tracker PRO', 'start_time': '2025-05-09T18:25:27.446417Z', 'end_time': '2025-05-09T18:25:39.842781Z', 'is_productive': False, 'app_name': 'python.exe', 'keyboard_presses': 0}
2025-05-09 21:25:49,453 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":345,"user":3,"application":21,"start_time":"2025-05-09T21:25:27.446417+03:00","end_time":"2025-05-09T21:25:39.842781+03:00","duration":"00:00:12.396364","keyboard_presses":0}
2025-05-09 21:25:49,454 - TimeTracker - INFO - Отправляем пайлоад: {'application': 21, 'title': 'Настройки отслеживания', 'start_time': '2025-05-09T18:25:39.843498Z', 'end_time': '2025-05-09T18:25:47.849504Z', 'is_productive': False, 'app_name': 'python.exe', 'keyboard_presses': 0}
2025-05-09 21:25:49,465 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":346,"user":3,"application":21,"start_time":"2025-05-09T21:25:39.843498+03:00","end_time":"2025-05-09T21:25:47.849504+03:00","duration":"00:00:08.006006","keyboard_presses":0}
2025-05-09 21:25:49,465 - TimeTracker - INFO - Успешно отправлено 2 записей активности.
2025-05-09 21:25:49,891 - TimeTracker - INFO - Добавлено 0 нажатий клавиш в активность
2025-05-09 21:25:49,938 - TimeTracker - INFO - Завершена сессия активности: App='python.exe', Title='Time Tracker PRO, Duration=2s. В очереди: 1
2025-05-09 21:25:49,984 - TimeTracker - INFO - Начата новая сессия активности: App='Cursor.exe', Title='python manage.py runserver - T...'
2025-05-09 21:25:50,053 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:25:51,320 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:25:52,046 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:25:56,327 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:25:59,034 - TimeTracker - INFO - Добавлено 23 нажатий клавиш в активность
2025-05-09 21:25:59,034 - TimeTracker - INFO - Завершена сессия активности: App='Cursor.exe', Title='python manage.py runserver - T..., Duration=9s. В очереди: 2
2025-05-09 21:25:59,440 - TimeTracker - INFO - Вычисленная длительность: 2.087793 секунд
2025-05-09 21:25:59,440 - TimeTracker - INFO - Для приложения python.exe выбран ID=21
2025-05-09 21:25:59,440 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:25:47.850352Z, end_time=2025-05-09T18:25:49.938145Z, длительность=2.087793 секунд
2025-05-09 21:25:59,441 - TimeTracker - INFO - Вычисленная длительность: 9.050312 секунд
2025-05-09 21:25:59,441 - TimeTracker - INFO - Отправляем запрос на создание приложения: http://127.0.0.1:8000/api/applications/
2025-05-09 21:25:59,450 - TimeTracker - INFO - Создано новое приложение: Cursor.exe с ID=23
2025-05-09 21:25:59,450 - TimeTracker - INFO - Для приложения Cursor.exe выбран ID=23
2025-05-09 21:25:59,450 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:25:49.984000Z, end_time=2025-05-09T18:25:59.034312Z, длительность=9.050312 секунд
2025-05-09 21:25:59,450 - TimeTracker - INFO - Отправка 2 записей активности на сервер.
2025-05-09 21:25:59,450 - TimeTracker - INFO - API URL: http://127.0.0.1:8000/api/activities/, Токен: eyJhbGciOi...
2025-05-09 21:25:59,450 - TimeTracker - INFO - Отправляем пайлоад: {'application': 21, 'title': 'Time Tracker PRO', 'start_time': '2025-05-09T18:25:47.850352Z', 'end_time': '2025-05-09T18:25:49.938145Z', 'is_productive': False, 'app_name': 'python.exe', 'keyboard_presses': 0}
2025-05-09 21:25:59,460 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":347,"user":3,"application":21,"start_time":"2025-05-09T21:25:47.850352+03:00","end_time":"2025-05-09T21:25:49.938145+03:00","duration":"00:00:02.087793","keyboard_presses":0}
2025-05-09 21:25:59,460 - TimeTracker - INFO - Отправляем пайлоад: {'application': 23, 'title': 'python manage.py runserver - Tracker33 - Cursor', 'start_time': '2025-05-09T18:25:49.984000Z', 'end_time': '2025-05-09T18:25:59.034312Z', 'is_productive': True, 'app_name': 'Cursor.exe', 'keyboard_presses': 23}
2025-05-09 21:25:59,469 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":348,"user":3,"application":23,"start_time":"2025-05-09T21:25:49.984000+03:00","end_time":"2025-05-09T21:25:59.034312+03:00","duration":"00:00:09.050312","keyboard_presses":23}
2025-05-09 21:25:59,470 - TimeTracker - INFO - Успешно отправлено 2 записей активности.
2025-05-09 21:25:59,898 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:26:01,572 - TimeTracker - INFO - Токен действителен еще 0.6 часов
2025-05-09 21:26:02,008 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:26:02,037 - TimeTracker - INFO - Начата новая сессия активности: App='chrome.exe', Title='Статистика - Time Tracker - Go...'
2025-05-09 21:26:02,334 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:26:04,039 - TimeTracker - INFO - Добавлено 0 нажатий клавиш в активность
2025-05-09 21:26:04,040 - TimeTracker - INFO - Завершена сессия активности: App='chrome.exe', Title='Статистика - Time Tracker - Go..., Duration=2s. В очереди: 1
2025-05-09 21:26:04,041 - TimeTracker - INFO - Начата новая сессия активности: App='chrome.exe', Title='Главная - Time Tracker - Googl...'
2025-05-09 21:26:07,341 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:26:09,046 - TimeTracker - INFO - Добавлено 0 нажатий клавиш в активность
2025-05-09 21:26:09,046 - TimeTracker - INFO - Завершена сессия активности: App='chrome.exe', Title='Главная - Time Tracker - Googl..., Duration=5s. В очереди: 2
2025-05-09 21:26:09,049 - TimeTracker - INFO - Начата новая сессия активности: App='chrome.exe', Title='Статистика - Time Tracker - Go...'
2025-05-09 21:26:09,446 - TimeTracker - INFO - Вычисленная длительность: 2.003117 секунд
2025-05-09 21:26:09,446 - TimeTracker - INFO - Для приложения chrome.exe выбран ID=22
2025-05-09 21:26:09,446 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:26:02.036974Z, end_time=2025-05-09T18:26:04.040091Z, длительность=2.003117 секунд
2025-05-09 21:26:09,447 - TimeTracker - INFO - Вычисленная длительность: 5.005683 секунд
2025-05-09 21:26:09,447 - TimeTracker - INFO - Для приложения chrome.exe выбран ID=22
2025-05-09 21:26:09,447 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:26:04.040919Z, end_time=2025-05-09T18:26:09.046602Z, длительность=5.005683 секунд
2025-05-09 21:26:09,447 - TimeTracker - INFO - Отправка 2 записей активности на сервер.
2025-05-09 21:26:09,447 - TimeTracker - INFO - API URL: http://127.0.0.1:8000/api/activities/, Токен: eyJhbGciOi...
2025-05-09 21:26:09,447 - TimeTracker - INFO - Отправляем пайлоад: {'application': 22, 'title': 'Статистика - Time Tracker - Google Chrome', 'start_time': '2025-05-09T18:26:02.036974Z', 'end_time': '2025-05-09T18:26:04.040091Z', 'is_productive': False, 'app_name': 'chrome.exe', 'keyboard_presses': 0}
2025-05-09 21:26:09,458 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":349,"user":3,"application":22,"start_time":"2025-05-09T21:26:02.036974+03:00","end_time":"2025-05-09T21:26:04.040091+03:00","duration":"00:00:02.003117","keyboard_presses":0}
2025-05-09 21:26:09,458 - TimeTracker - INFO - Отправляем пайлоад: {'application': 22, 'title': 'Главная - Time Tracker - Google Chrome', 'start_time': '2025-05-09T18:26:04.040919Z', 'end_time': '2025-05-09T18:26:09.046602Z', 'is_productive': False, 'app_name': 'chrome.exe', 'keyboard_presses': 0}
2025-05-09 21:26:09,468 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":350,"user":3,"application":22,"start_time":"2025-05-09T21:26:04.040919+03:00","end_time":"2025-05-09T21:26:09.046602+03:00","duration":"00:00:05.005683","keyboard_presses":0}
2025-05-09 21:26:09,468 - TimeTracker - INFO - Успешно отправлено 2 записей активности.
2025-05-09 21:26:10,083 - TimeTracker - INFO - Добавлено 0 нажатий клавиш в активность
2025-05-09 21:26:10,095 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:26:10,095 - TimeTracker - INFO - Завершена сессия активности: App='chrome.exe', Title='Статистика - Time Tracker - Go..., Duration=1s. В очереди: 1
2025-05-09 21:26:10,097 - TimeTracker - INFO - Начата новая сессия активности: App='chrome.exe', Title='Главная - Time Tracker - Googl...'
2025-05-09 21:26:11,098 - TimeTracker - INFO - Добавлено 0 нажатий клавиш в активность
2025-05-09 21:26:11,098 - TimeTracker - INFO - Завершена сессия активности: App='chrome.exe', Title='Главная - Time Tracker - Googl..., Duration=1s. В очереди: 2
2025-05-09 21:26:11,100 - TimeTracker - INFO - Начата новая сессия активности: App='chrome.exe', Title='Логи - Time Tracker - Google C...'
2025-05-09 21:26:12,042 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:26:12,347 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:26:17,355 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:26:19,453 - TimeTracker - INFO - Токен действителен еще 0.6 часов
2025-05-09 21:26:19,454 - TimeTracker - INFO - Вычисленная длительность: 1.047551 секунд
2025-05-09 21:26:19,454 - TimeTracker - INFO - Для приложения chrome.exe выбран ID=22
2025-05-09 21:26:19,454 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:26:09.047958Z, end_time=2025-05-09T18:26:10.095509Z, длительность=1.047551 секунд
2025-05-09 21:26:19,454 - TimeTracker - INFO - Вычисленная длительность: 1.00177 секунд
2025-05-09 21:26:19,455 - TimeTracker - INFO - Для приложения chrome.exe выбран ID=22
2025-05-09 21:26:19,455 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:26:10.096663Z, end_time=2025-05-09T18:26:11.098433Z, длительность=1.00177 секунд
2025-05-09 21:26:19,455 - TimeTracker - INFO - Отправка 2 записей активности на сервер.
2025-05-09 21:26:19,455 - TimeTracker - INFO - API URL: http://127.0.0.1:8000/api/activities/, Токен: eyJhbGciOi...
2025-05-09 21:26:19,455 - TimeTracker - INFO - Отправляем пайлоад: {'application': 22, 'title': 'Статистика - Time Tracker - Google Chrome', 'start_time': '2025-05-09T18:26:09.047958Z', 'end_time': '2025-05-09T18:26:10.095509Z', 'is_productive': False, 'app_name': 'chrome.exe', 'keyboard_presses': 0}
2025-05-09 21:26:19,465 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":351,"user":3,"application":22,"start_time":"2025-05-09T21:26:09.047958+03:00","end_time":"2025-05-09T21:26:10.095509+03:00","duration":"00:00:01.047551","keyboard_presses":0}
2025-05-09 21:26:19,465 - TimeTracker - INFO - Отправляем пайлоад: {'application': 22, 'title': 'Главная - Time Tracker - Google Chrome', 'start_time': '2025-05-09T18:26:10.096663Z', 'end_time': '2025-05-09T18:26:11.098433Z', 'is_productive': False, 'app_name': 'chrome.exe', 'keyboard_presses': 0}
2025-05-09 21:26:19,476 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":352,"user":3,"application":22,"start_time":"2025-05-09T21:26:10.096663+03:00","end_time":"2025-05-09T21:26:11.098433+03:00","duration":"00:00:01.001770","keyboard_presses":0}
2025-05-09 21:26:19,476 - TimeTracker - INFO - Успешно отправлено 2 записей активности.
2025-05-09 21:26:19,889 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:26:22,057 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:26:22,362 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:26:27,369 - TimeTracker - ERROR - Ошибка в потоке отслеживания активности: 'TimeTrackerApp' object has no attribute 'is_app_useful'
Traceback (most recent call last):
  File "C:\Users\mihai\Heist_master_PC\Documents\GitHub\Tracker33\desktop_app\main.py", line 671, in track_activity
    # Если приложение изменилось
AttributeError: 'TimeTrackerApp' object has no attribute 'is_app_useful'
2025-05-09 21:26:28,110 - TimeTracker - INFO - Добавлено 0 нажатий клавиш в активность
2025-05-09 21:26:28,110 - TimeTracker - INFO - Завершена сессия активности: App='chrome.exe', Title='Логи - Time Tracker - Google C..., Duration=17s. В очереди: 1
2025-05-09 21:26:28,111 - TimeTracker - INFO - Начата новая сессия активности: App='Cursor.exe', Title='python manage.py runserver - T...'
2025-05-09 21:26:29,454 - TimeTracker - INFO - Вычисленная длительность: 17.011105 секунд
2025-05-09 21:26:29,455 - TimeTracker - INFO - Для приложения chrome.exe выбран ID=22
2025-05-09 21:26:29,455 - TimeTracker - INFO - Отправка активности: start_time=2025-05-09T18:26:11.099363Z, end_time=2025-05-09T18:26:28.110468Z, длительность=17.011105 секунд
2025-05-09 21:26:29,455 - TimeTracker - INFO - Отправка 1 записей активности на сервер.
2025-05-09 21:26:29,455 - TimeTracker - INFO - API URL: http://127.0.0.1:8000/api/activities/, Токен: eyJhbGciOi...
2025-05-09 21:26:29,455 - TimeTracker - INFO - Отправляем пайлоад: {'application': 22, 'title': 'Логи - Time Tracker - Google Chrome', 'start_time': '2025-05-09T18:26:11.099363Z', 'end_time': '2025-05-09T18:26:28.110468Z', 'is_productive': False, 'app_name': 'chrome.exe', 'keyboard_presses': 0}
2025-05-09 21:26:29,465 - TimeTracker - INFO - Успешно отправлено: 201 - {"id":353,"user":3,"application":22,"start_time":"2025-05-09T21:26:11.099363+03:00","end_time":"2025-05-09T21:26:28.110468+03:00","duration":"00:00:17.011105","keyboard_presses":0}
2025-05-09 21:26:29,466 - TimeTracker - INFO - Успешно отправлено 1 записей активности.
2025-05-09 21:26:30,086 - TimeTracker - INFO - Обновлен список приложений: 107 процессов
2025-05-09 21:26:31,573 - TimeTracker - INFO - Токен действителен еще 0.6 часов
